### PR TITLE
Clarify supported platform in our docs

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
This will add clarification on the Kubernetes platforms we support and test. This is somewhat present on https://www.scylladb.com/product/scylladb-operator-kubernetes/ but not very well updated or detailed.

Screenshot from 2024-04-03 17-16-40

### CODE REVIEW

File: docs/source/releases.md [hunk 1]
```diff
@@ -67,3 +67,30 @@ Support matrix table shows the version requirements for a particular **scylla-op
...
+| EKS              | Amazon Linux |
+:::
+
+While our APIs generally work on any Kubernetes conformant cluster,
+performance tuning and other pieces that need to interact with the host OS, kubelet, cri, kernel, etc. might hit some incompatibilities.
+
+
+:::{warning}
+The following platforms are known **not to work correctly** at this time.
+This may change in the future.
```
Comment (rzetelskik) [inline docs/source/releases.md:88]: nit: imo that's redundant and also against [style recommendations ](https://docs.google.com/document/d/1lyHp1MKdyj0Hh3NprNFvEczA4dFSZIFoukGUvFJb9yE)

image

File: docs/source/releases.md [hunk 2]
```diff
@@ -67,3 +67,30 @@ Support matrix table shows the version requirements for a particular **scylla-op
...
+
+:::{table}
+| Platform         | OS Image     |
+|:-----------------|:-------------|
+| GKE              | Ubuntu       |
+| EKS              | Amazon Linux |
+:::
+
+While our APIs generally work on any Kubernetes conformant cluster,
+performance tuning and other pieces that need to interact with the host OS, kubelet, cri, kernel, etc. might hit some incompatibilities.
```
Comment (zimnx) [inline docs/source/releases.md:83]:
```suggestion
performance tuning and other pieces that need to interact with the host OS, kubelet, CRI, kernel, etc. might hit some incompatibilities.
```

File: docs/source/releases.md [hunk 3]
```diff
@@ -67,3 +67,30 @@ Support matrix table shows the version requirements for a particular **scylla-op
 | Scylla Manager    | `>=3.2.6`  | `>=3.2.6`  | `>=3.2`    | `>=2.6`    |
 | Scylla Monitoring | `(CRD)`    | `(CRD)`    | `(CRD)`    | `>=4.0`    |
 :::
+
+### Supported Kubernetes platforms
+
+We officially test and recommend to use the following platforms.
```
[the remark this answers is not in the document]
Comment (tnozicka) [reply docs/source/releases.md]: fixed, thx

File: docs/source/releases.md (hunk 1 above)
Comment (tnozicka) [reply docs/source/releases.md]: removed

File: docs/source/releases.md (hunk 2 above)
Comment (tnozicka) [reply docs/source/releases.md]: fixed, thx

**Review by rzetelskik — APPROVED** [review]
lgtm
/assign zimnx

**Pull request merged**